### ISSUE TITLE
Proposition off small additions/fixes for topologicpy

### ISSUE THREAD

**Issue by Sinasta** [wassimj/topologicpy#11]
I am maintaining [a fork of topologicpy](https://github.com/Sinasta/topologicpy) with some features/fixes for my personal project:
Those changes are:
- [implementing progressbars (tqdm) for DGL](https://github.com/wassimj/topologicpy/commit/192214b03b54cc70dbf9f3d39fd553c33d86b351)
- [make DGL.Show() return a Plotly figure](https://github.com/wassimj/topologicpy/commit/dda04387f1f8fb741238c7cb7bf98023a8fd60bb)
- [adapt EnergyModel.py to new topologicpy API](https://github.com/wassimj/topologicpy/commit/ef6575d362bcdcf53301f943e62b43e861aa2756)
- [remove info prints in EnergyModel.py](https://github.com/wassimj/topologicpy/commit/1501b84d3e78240070a8b28bd21cfeefa2494587)
- [temporary personal fix because of outdated topologic (opencascade) version](https://github.com/wassimj/topologicpy/commit/e1c9177f15818568c554c8da017df1d800cabcec)

I thought you could be interested in some of those.
I will close this issue because it is actually not an issue.
P.s. if you think that adding progress-bars to DGL is a good idea but don't want to rely on an external library (tqdm) then i would be happy to propose a simple native python progress print.

**Comment by wassimj** [comment]
Funny.. I just implemented tqdm for importing OBJ
DGL.Show: I will add the same as Topology  (Plotly.DataByDGL, Plotly.FigureByDGLData)
EnergyModel needs refactoring work
topologic inside pip install topologicpy is the latest. It uses opencascade 7.7.0

**Comment by wassimj** [comment]
Please see new push to github regarding "**_make DGL.Show() return a Plotly figure_**"   You can now replace that with:

```
data = DGL.Train(hparams=hparams, trainingDataset=traintest_ds, validationDataset=None, overwrite=True)
labels = ["Epochs", "Testing Accuracy", "Training Accuracy", "Testing Loss", "Training Loss"]
df = Plotly.DataByDGL(data=data, labels=labels)
fig = Plotly.FigureByDataFrame(df,
                         labels=labels,
                         title="Untitled",
                         x_title="Epochs",
                         x_spacing=1.0,
                         y_title="Accuracy and Loss",
                         y_spacing=0.1,
                         use_markers=False,
                         chart_type="Line",
                         renderer = "notebook")
Plotly.Show(fig, renderer="notebook")

```

**Comment by Sinasta** [comment]
Awesome! That's indeed more coherent then my approach.

**Comment by wassimj** [comment]
@Sinasta Is the EnergyModel functional and tested? If so, please consider making a PR on the main branch. Thanks.

**Comment by Sinasta** [comment]
I would be happy to propose a PR but i will do some testing before.
What do you think about removing the prints in EnergyModel.py?
As in this commit:
[remove info prints in EnergyModel.py](https://github.com/wassimj/topologicpy/commit/1501b84d3e78240070a8b28bd21cfeefa2494587)
Maybe you prefer to keep those prints or make them optional with an argument?
I am suppressing as well the EnergyPlus warnings with this line:
`openstudio.Logger.instance().standardOutLogger().setLogLevel(openstudio.Fatal)`
Maybe you know a better option?
I was wanting to suppress the warnings because it confused my tqdm Bar.

**Comment by wassimj** [comment]
Yeah no problem. EnergyModel did not get any attention from me for a long time. Thank you.

**Comment by wassimj** [comment]
Hi @Sinasta,  tqdm has two flavours (regular and notebook). topologicpy can be used in either modes. Does or can your customised DGL code accommodate both modes? Perhaps there needs to be an additional parameter to the DGL methods that use tqdm? (I call that parameter "**renderer**" and the values can be "**browser**" or "**notebook**" and the default is "**notebook**" Let me know. If you merge the latest commit to your fork and then test in jupyter notebook as well as from a command prompt  and all is well, please issue a PR on DGL. Happy to include it with the next release. Thank you again!

**Comment by Sinasta** [comment]
The adaptivity of tqdm to the environment (ipykernel or terminal) is easily handled with:
`from tqdm.auto import tqdm`
I will happily propose a PR today for integrating tqdm in DGL.

**Comment by wassimj** [comment]
Great. But please pause that as I am about to publish a new version where I streamlined and added a few more options to Plotly and DGL files. This will be version 0.2.8 (very soon hopefully. Just testing now)

**Comment by Sinasta** [comment]
Sorry i saw your message to late. I will redo the PR as soon as you publish the new version.

**Comment by wassimj** [comment]
Yes, no worries. It has all been published. I am now on version 0.2.8

**Comment by Sinasta** [comment]
Perfect i opened a new PR for the tqdm bars

**Comment by wassimj** [comment]
These have now been published in version 0.2.9. Thank you for all your help!
